scripts/ci/verify_required_ci_run.py

```
@@ -0,0 +1,193 @@
+    try:
+        completed = subprocess.run(
+            ["gh", "api", endpoint],
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            env=os.environ.copy(),
+        )
+    except (OSError, subprocess.CalledProcessError) as error:
+        detail = getattr(error, "stderr", "") or str(error)
+        raise RequiredCIVerificationError(f"GitHub API request failed: {detail.strip()}") from error
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)